tests/drivers/udc/src/main.c

```diff
@@ -352,14 +352,22 @@ static void test_udc_ep_api(const struct device *dev,
 static void test_udc_ep_mps(uint8_t type)
 {
 	uint16_t mps[] = {8, 16, 32, 64, 512, 1024};
-	struct usb_ep_descriptor ed = {
+	struct usb_ep_descriptor ed_out = {
 		.bLength = sizeof(struct usb_ep_descriptor),
 		.bDescriptorType = USB_DESC_ENDPOINT,
 		.bEndpointAddress = 0x01,
 		.bmAttributes = type,
 		.wMaxPacketSize = sys_cpu_to_le16(0),
 		.bInterval = 0,
 	};
+	struct usb_ep_descriptor ed_in = {
```

Comment:
Actually, you do not need another endpoint descriptor because you are passing `i | USB_EP_DIR_IN` argument anyway. I also had to fix it, but in a slightly different way, see https://github.com/zephyrproject-rtos/zephyr/pull/94766/commits/d2221db2f40448335d30f2e17b0744453244c2ef#diff-e7bf79a7d5fa6d4a560009ffc9f0f10b977bc121b24b166809b198c4b30a2dfb